src/lib/parsers.ts

```diff
@@ -0,0 +1,238 @@
+import {
+  // Subtypes
+  Team,
+  Maiden,
+  GameMap,
+  CabOrientation,
+  VictoryType,
+  // Event types
+  PlayerNames,
+  PlayerKill,
+  BlessMaiden,
+  ReserveMaiden,
+  UnreserveMaiden,
+  UseMaiden,
+  Glance,
+  CarryFood,
+  GameStart,
+  GameEnd,
+  Victory,
+  Spawn,
+  GetOnSnail,
+  GetOffSnail,
+  SnailEat,
+  SnailEscape,
+  BerryDeposit,
+  BerryKickIn,
+} from './models/KQStream';
+
+const teams = {
+  'Red': Team.Gold,
+  'Gold': Team.Gold,
+  'Blue': Team.Blue,
+};
+
+const maidens = {
+  'maiden_wings': Maiden.Warrior,
+  'maiden_speed': Maiden.Speed,
+};
+
+const maps = {
+  'map_day': GameMap.Day,
+  'map_night': GameMap.Night,
+  'map_dusk': GameMap.Dusk,
+};
+
+const orientations = {
+  'False': CabOrientation.BlueGold,
+  'True': CabOrientation.GoldBlue,
+};
+
+const victories = {
+  'military': VictoryType.Military,
+  'economic': VictoryType.Econimic,
+  'snail': VictoryType.Snail,
+};
+
+const booleans = {
+  'False': false,
+  'True': true,
+};
+
+export function playernames(value: string): PlayerNames {
+  // Not sure what the values of the message mean,
+  // so just return an empty object for now.
+  return {};
+}
+
+export function playerKill(value: string): PlayerKill {
+  const [x, y, by, killed] = value.split(',');
+  return {
+    pos: {
```

Comment:
You don't have to do this now, but it would be nice to move this code to a function, to reduce repetition.